Hivemind/Assets/Scripts/Gathering.cs

```diff
@@ -97,42 +168,38 @@ private void Update()
                 {
                     if (nextHarvest > 0)
                     {
-                        nextHarvest--;
-                        break;
-                    }
-                    target = findResource();
-                    if (target != null)
-                    {
-                        nextHarvest = target.DecreaseFutureResources(CarryAmount - inventoryAmount);
-                        agent.SetDestination(target.GetPosition());
-                        state = State.MovingToResource;
-                    }
-                    else
-                    {
-                        state = State.MovingToStorage;
                         break;
                     }
+                    TargetResource();
                 }
                 break;
+
             case State.MovingToStorage:
                 if (storage != null)
                 {
                     agent.SetDestination(storage.GetPosition());
-                    for(int i = 0; i < carryingObjects.Count; i++)
+                    for (int i = 0; i < carryingObjects.Count; i++)
                     {
                         carryingObjects[i].transform.position = new Vector3(transform.position.x, transform.position.y + 0.08f * (i + 1), transform.position.z);
                     }
                     if (Vector3.Distance(transform.position, storage.GetPosition()) < 2f)
                     {
-                        GameResources.AddResources(inventory);
-                        inventory.Clear();
-                        inventoryAmount = 0;
-                        foreach(GameObject gameObject in carryingObjects)
+                        if (IsScout && target != null)
+                        {
+                            target.AddToKnownResourceList();
+                        }
+                        if (inventory != null)
                         {
-                            Destroy(gameObject);
+                            GameResources.AddResources(inventory);
+                            inventory.Clear();
+                            inventoryAmount = 0;
```

Comment:
waar is deze waarde voor aangezien je die kan berekenen uit het aantal elementen in carryingObjects.(count of size idk)

Reply:
Deze waarde was outdated, hij is nu vervangen door een count van de carryingObjects lijst